### PR TITLE
Use a patch for mysql-server instead of kyrofa's custom repository

### PR DESCRIPTION
**Pull request by pachulo**
This PR changes the repository from where mysql-server is pulled and adds kyrofa's patch to make it run in heavily confined systems like Ubuntu Core.

It also updates mysql to 5.7.22

This tries to solve #329

### CODE REVIEW

**Comment by pachulo** [comment]
It looks like it built and passed the test, but in the log file (https://circleci.com/api/v1.1/project/github/nextcloud/nextcloud-snap/962/output/102/0?file=true) I can see these warnings:

```
CMake Warning:

  Manually-specified variables were not used by the project:



    WITH_EMBEDDED_SHARED_LIBRARY

    WITH_INNODB_PAGE_CLEANER_PRIORITY
```

What do you think @kyrofa ?

**Comment by pachulo** [comment]
Well, tried the refresh with the updated mysql in my own instance and everything seems to be working fine!

Here are the builds: https://code.launchpad.net/~pachulo/+snap/nextcloud-snap-mysql-patch

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -254,9 +254,11 @@ parts:
 
   mysql:
     plugin: cmake
-    source: https://github.com/kyrofa/mysql-server.git
-    source-branch: mysql-5.7.18_support_no_setpriority
-    after: [boost]
+    source: https://github.com/mysql/mysql-server
+    source-type: git
```
Comment (kyrofa) [inline snap/snapcraft.yaml]: If you use `https://github.com/mysql/mysql-server.git` you don't have to specify the source-type.

**Comment by pachulo** [comment]
OK, it seems that it worked now; cmake is only showing this warning:

```
CMake Warning:

  Manually-specified variables were not used by the project:



    WITH_EMBEDDED_SHARED_LIBRARY
```

**Comment by pachulo** [comment]
Done!

**Comment by pachulo** [comment]
Do you need anything else from me here @kyrofa ?

**Comment by kyrofa** [comment]
No, this looks excellent, I just want to give it some hefty testing and my last few weekends have been booked solid! This weekend is looking promising, though.

**Comment by kyrofa** [comment]
A snap of this PR has been pushed to the `beta/pr-581` channel.

**Comment by pachulo** [comment]
The refresh went well here.

**Comment by kyrofa** [comment]
You refreshed to it? From what? If a snap running an older mysql, can you find the upgrade log in your syslog (or mysql journal)? I have logic there to actually update the database, but it's never been exercised before now.

**Comment by pachulo** [comment]
Yeah, I refreshed from the stable channel one: revision 6924. 
Here you have the log:

```
Jun 08 21:37:53 localhost.localdomain nextcloud.mysql[28016]: ....... *
Jun 08 21:37:53 localhost.localdomain nextcloud.mysql[28016]: Checking/upgrading mysql tables if necessary...
Jun 08 21:37:53 localhost.localdomain nextcloud.php-fpm[28000]: Obtaining nextcloud mysql credentials... done
Jun 08 21:37:53 localhost.localdomain nextcloud.mysql[28016]: Checking if update is needed.
Jun 08 21:37:53 localhost.localdomain nextcloud.mysql[28016]: Checking server version.
Jun 08 21:37:53 localhost.localdomain nextcloud.mysql[28016]: Running queries to upgrade MySQL server.
Jun 08 21:38:24 localhost.localdomain nextcloud.mysql[28016]: Checking system database.
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.columns_priv                                 OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.db                                           OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.engine_cost                                  OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.event                                        OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.func                                         OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.general_log                                  OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.gtid_executed                                OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.help_category                                OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.help_keyword                                 OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.help_relation                                OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.help_topic                                   OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.innodb_index_stats                           OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.innodb_table_stats                           OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.ndb_binlog_index                             OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.plugin                                       OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.proc                                         OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.procs_priv                                   OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.proxies_priv                                 OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.server_cost                                  OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.servers                                      OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.slave_master_info                            OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.slave_relay_log_info                         OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.slave_worker_info                            OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.slow_log                                     OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.tables_priv                                  OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.time_zone                                    OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.time_zone_leap_second                        OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.time_zone_name                               OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.time_zone_transition                         OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.time_zone_transition_type                    OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: mysql.user                                         OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: The sys schema is already up to date (version 1.5.1).
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: Checking databases.
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_accounts                              OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_activity                              OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_activity_mq                           OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_addressbookchanges                    OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_addressbooks                          OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_admin_sections                        OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_admin_settings                        OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_appconfig                             OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_authtoken                             OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_bookmarks                             OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_bookmarks_tags                        OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_bruteforce_attempts                   OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_calendarchanges                       OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_calendarobjects                       OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_calendarobjects_props                 OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_calendars                             OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_calendarsubscriptions                 OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_cards                                 OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_cards_properties                      OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_comments                              OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_comments_read_markers                 OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_credentials                           OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_dav_shares                            OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_external_applicable                   OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_external_config                       OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_external_mounts                       OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_external_options                      OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_federated_reshares                    OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_file_locks                            OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_filecache                             OK
Jun 08 21:38:25 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_files_trash                           OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_flow_checks                           OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_flow_operations                       OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_group_admin                           OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_group_user                            OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_groups                                OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_jobs                                  OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_ldap_group_mapping                    OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_ldap_group_members                    OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_ldap_user_mapping                     OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_migrations                            OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_mimetypes                             OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_mounts                                OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_news_feeds                            OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_news_folders                          OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_news_items                            OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_notes_meta                            OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_notifications                         OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_notifications_pushtokens              OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_oauth2_access_tokens                  OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_oauth2_clients                        OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_personal_sections                     OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_personal_settings                     OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_preferences                           OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_privatedata                           OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_properties                            OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_schedulingobjects                     OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_share                                 OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_share_external                        OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_storages                              OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_systemtag                             OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_systemtag_group                       OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_systemtag_object_mapping              OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_trusted_servers                       OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_twofactor_backupcodes                 OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_users                                 OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_vcategory                             OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: nextcloud.oc_vcategory_to_object                   OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: sys.sys_config                                     OK
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: Upgrade process completed successfully.
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: Checking if update is needed.
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: Restarting mysql server after upgrade...
Jun 08 21:38:26 localhost.localdomain audit[28988]: AVC apparmor="DENIED" operation="exec" profile="snap.nextcloud.mysql" name="/bin/systemctl" pid=28988 comm="mysql.serve
r" requested_mask="x" denied_mask="x" fsuid=0 ouid=0
Jun 08 21:38:26 localhost.localdomain kernel: audit: type=1400 audit(1528493906.278:105): apparmor="DENIED" operation="exec" profile="snap.nextcloud.mysql" name="/bin/syst
emctl" pid=28988 comm="mysql.server" requested_mask="x" denied_mask="x" fsuid=0 ouid=0
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: my_print_defaults: [ERROR] Could not open required defaults file:  /snap/nextcloud/7452/my.cnf
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: my_print_defaults: [ERROR] Fatal error in defaults handling. Program aborted!
Jun 08 21:38:26 localhost.localdomain audit[29003]: AVC apparmor="DENIED" operation="exec" profile="snap.nextcloud.mysql" name="/bin/systemctl" pid=29003 comm="mysql.serve
r" requested_mask="x" denied_mask="x" fsuid=0 ouid=0
Jun 08 21:38:26 localhost.localdomain kernel: audit: type=1400 audit(1528493906.406:106): apparmor="DENIED" operation="exec" profile="snap.nextcloud.mysql" name="/bin/syst
emctl" pid=29003 comm="mysql.server" requested_mask="x" denied_mask="x" fsuid=0 ouid=0
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: my_print_defaults: [ERROR] Could not open required defaults file:  /snap/nextcloud/7452/my.cnf
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: my_print_defaults: [ERROR] Fatal error in defaults handling. Program aborted!
Jun 08 21:38:26 localhost.localdomain nextcloud.mysql[28016]: Shutting down MySQL
Jun 08 21:38:32 localhost.localdomain nextcloud.mysql[28016]: ...... *
Jun 08 21:38:32 localhost.localdomain audit[29033]: AVC apparmor="DENIED" operation="exec" profile="snap.nextcloud.mysql" name="/bin/systemctl" pid=29033 comm="mysql.serve
r" requested_mask="x" denied_mask="x" fsuid=0 ouid=0
Jun 08 21:38:32 localhost.localdomain kernel: audit: type=1400 audit(1528493912.606:107): apparmor="DENIED" operation="exec" profile="snap.nextcloud.mysql" name="/bin/syst
emctl" pid=29033 comm="mysql.server" requested_mask="x" denied_mask="x" fsuid=0 ouid=0
Jun 08 21:38:32 localhost.localdomain nextcloud.mysql[28016]: my_print_defaults: [ERROR] Could not open required defaults file:  /snap/nextcloud/7452/my.cnf
Jun 08 21:38:32 localhost.localdomain nextcloud.mysql[28016]: my_print_defaults: [ERROR] Fatal error in defaults handling. Program aborted!
Jun 08 21:38:32 localhost.localdomain nextcloud.mysql[28016]: Starting MySQL
Jun 08 21:38:38 localhost.localdomain nextcloud.mysql[28016]: ...... *

```

**Pull request merged**